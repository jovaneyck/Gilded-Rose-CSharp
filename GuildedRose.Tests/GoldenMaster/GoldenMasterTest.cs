using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using GuildedRose.Console;
using Newtonsoft.Json;
using Xunit;

namespace GuildedRose.Tests.GoldenMaster
{
    [UseReporter(typeof(XUnit2Reporter))]
    //[UseReporter(typeof(DiffReporter))]
    public class GoldenMasterTest
    {
        [Fact]
        public void RandomInputKeepsSameResultsAsBefore()
        {
            var items = new[]
            {
                new Item {Name = "_item_name_", Quality = 10, SellIn = 30},
                new Item {Name = "Aged Brie", Quality = 10, SellIn = 30},
                new Item {Name= "Backstage passes to a TAFKAL80ETC concert", Quality = 10, SellIn = 10 },
                new Item {Name= "Backstage passes to a TAFKAL80ETC concert", Quality = 10, SellIn = 5 },
                new Item {Name= "Sulfuras, Hand of Ragnaros", Quality = 10, SellIn = 5 },
                new Item {Name = "Aged Brie", Quality = 10, SellIn = 0},
                new Item {Name = "Backstage passes to a TAFKAL80ETC concert", Quality = 3, SellIn = 0},
                new Item {Name = "_an_expired_item_", Quality = 3, SellIn = 0},
                new Item {Name = "Sulfuras, Hand of Ragnaros", Quality = 3, SellIn = 0}

            };

            var app = new Program { Items = items};

            app.UpdateQuality();

            Approvals.VerifyJson(JsonConvert.SerializeObject(app.Items));
        } 
    }
}