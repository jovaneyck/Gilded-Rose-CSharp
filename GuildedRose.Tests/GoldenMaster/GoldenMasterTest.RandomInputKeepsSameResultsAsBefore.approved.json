[
    {
        "Name": "_item_name_",
        "SellIn": 29,
        "Quality": 9
    },
    {
        "Name": "Aged Brie",
        "SellIn": 29,
        "Quality": 11
    },
    {
        "Name": "Backstage passes to a TAFKAL80ETC concert",
        "SellIn": 9,
        "Quality": 12
    },
    {
        "Name": "Backstage passes to a TAFKAL80ETC concert",
        "SellIn": 4,
        "Quality": 13
    },
    {
        "Name": "Sulfuras, Hand of Ragnaros",
        "SellIn": 5,
        "Quality": 10
    },
    {
        "Name": "Aged Brie",
        "SellIn": -1,
        "Quality": 12
    },
    {
        "Name": "Backstage passes to a TAFKAL80ETC concert",
        "SellIn": -1,
        "Quality": 0
    },
    {
        "Name": "_an_expired_item_",
        "SellIn": -1,
        "Quality": 1
    },
    {
        "Name": "Sulfuras, Hand of Ragnaros",
        "SellIn": 0,
        "Quality": 3
    }
]